tools/board-runner/src/esp32_c3.rs

```diff
@@ -0,0 +1,328 @@
+
+    let app = OpenOptions::new()
+        .append(true)
+        .create(false)
+        .open("app")
+        .unwrap();
+
+    let mut build = Command::new("cat")
+        .arg(format!(
+            "{}/{}",
+            env::var("LIBTOCK_C_TREE").unwrap(),
+            "examples/buttons/build/rv32imac/rv32imac.0x403B0060.0x3FCC0000.tbf"
+        ))
+        .stdout(app)
+        .spawn()
+        .expect("failed to spawn build");
+    assert!(build.wait().unwrap().success());
+
+    let mut p = esp32_c3_flash("../../tools/board-runner/app").unwrap();
+
+    p.exp_string("Hello World!")?;
+
+    println!("Make sure the LEDs are flashing");
+
+    let timeout = time::Duration::from_secs(10);
+    thread::sleep(timeout);
+
+    Ok(())
+}
+
+fn esp32_c3_malloc_test1() -> Result<(), Error> {
+    let app = format!(
+        "{}/{}",
+        env::var("LIBTOCK_C_TREE").unwrap(),
+        "examples/tests/malloc_test01/build/rv32imac/rv32imac.0x403B0060.0x3FCC0000.tbf"
+    );
+    let mut p = esp32_c3_flash(&app).unwrap();
+
+    p.exp_string("malloc01: success")?;
+
+    Ok(())
+}
+
+#[allow(dead_code)]
+fn esp32_c3_stack_size_test1() -> Result<(), Error> {
+    let app = format!(
+        "{}/{}",
+        env::var("LIBTOCK_C_TREE").unwrap(),
+        "examples/tests/stack_size_test01/build/rv32imac/rv32imac.0x403B0060.0x3FCC0000.tbf"
+    );
+    let mut p = esp32_c3_flash(&app).unwrap();
+
+    p.exp_string("Stack Test App")?;
+    p.exp_string("Current stack pointer: 0x3fcc0368")?;
+
+    Ok(())
+}
+
+#[allow(dead_code)]
+fn esp32_c3_stack_size_test2() -> Result<(), Error> {
+    let app = format!(
+        "{}/{}",
+        env::var("LIBTOCK_C_TREE").unwrap(),
+        "examples/tests/stack_size_test02/build/rv32imac/rv32imac.0x403B0060.0x3FCC0000.tbf"
+    );
+    let mut p = esp32_c3_flash(&app).unwrap();
+
+    p.exp_string("Stack Test App")?;
+    p.exp_string("Current stack pointer: 0x3fcc2310")?;
+
+    Ok(())
+}
+
+#[allow(dead_code)]
+fn esp32_c3_mpu_stack_growth() -> Result<(), Error> {
+    let app = format!(
+        "{}/{}",
+        env::var("LIBTOCK_C_TREE").unwrap(),
+        "examples/tests/mpu_stack_growth/build/rv32imac/rv32imac.0x403B0060.0x3FCC0000.tbf"
+    );
+    let mut p = esp32_c3_flash(&app).unwrap();
+
+    p.exp_string("This test should recursively add stack frames until exceeding")?;
+    p.exp_string("panicked at 'Process mpu_stack_growth had a fault'")?;
+    p.exp_string("EXCEEDED!")?;
+
+    Ok(())
+}
+
+#[allow(dead_code)]
+fn esp32_c3_mpu_walk_region() -> Result<(), Error> {
+    let app = format!(
+        "{}/{}",
+        env::var("LIBTOCK_C_TREE").unwrap(),
+        "examples/tests/mpu_walk_region/build/rv32imac/rv32imac.0x403B0060.0x3FCC0000.tbf"
+    );
+    let mut p = esp32_c3_flash(&app).unwrap();
+
+    p.exp_string("MPU Walk Regions")?;
+    p.exp_string("Walking flash")?;
+    // p.exp_string("Will overrun")?;
+    // p.exp_string("panicked at 'Process mpu_walk_region had a fault'")?;
```

Comment:
```suggestion
```
No need for these since the board does not have a button attached, right?

Reply:
The board has a physical button. it's not attached in Tock though

Reply:
gotcha, makes sens to keep them then